api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsWithHighViralLoadCohortDataset.java

```diff
@@ -0,0 +1,526 @@
+        listOfPatientsWithHighViralLoadCohortQueries;
+    this.listOfPatientsArtCohortCohortQueries = listOfPatientsArtCohortCohortQueries;
+    this.tptListOfPatientsEligibleDataSet = tptListOfPatientsEligibleDataSet;
+    this.tptInitiationDataDefinitionQueries = tptInitiationDataDefinitionQueries;
+    this.listOfPatientsDefaultersOrIITCohortQueries = listOfPatientsDefaultersOrIITCohortQueries;
+    this.listChildrenOnARTandFormulationsDataset = listChildrenOnARTandFormulationsDataset;
+  }
+
+  public DataSetDefinition contructDataset() {
+
+    PatientDataSetDefinition pdd = new PatientDataSetDefinition();
+
+    pdd.setName("HIGHVL");
+
+    PatientIdentifierType identifierType =
+        Context.getPatientService()
+            .getPatientIdentifierTypeByUuid("e2b966d0-1d5f-11e0-b929-000c29ad1d07");
+
+    PersonAttributeType contactAttributeType =
+        Context.getPersonService()
+            .getPersonAttributeTypeByUuid("e2e3fd64-1d5f-11e0-b929-000c29ad1d07");
+
+    DataDefinition conctactDef =
+        new ConvertedPersonDataDefinition(
+            "contact",
+            new PersonAttributeDataDefinition(contactAttributeType.getName(), contactAttributeType),
+            null);
+
+    PersonAttributeType referencePersonContactAttributeType =
+        Context.getPersonService()
+            .getPersonAttributeTypeByUuid("4d9d4863-6637-4aba-a907-f2e60f5be9c0");
+
+    DataDefinition referenceConctactDef =
+        new ConvertedPersonDataDefinition(
+            "reference",
+            new PersonAttributeDataDefinition(
+                referencePersonContactAttributeType.getName(), referencePersonContactAttributeType),
+            null);
+
+    DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
+
+    DataDefinition identifierDef =
+        new ConvertedPatientDataDefinition(
+            "identifier",
+            new PatientIdentifierDataDefinition(identifierType.getName(), identifierType),
+            identifierFormatter);
+
+    DataConverter formatter = new ObjectFormatter("{familyName}, {givenName}");
+
+    DataDefinition nameDef =
+        new ConvertedPersonDataDefinition("name", new PreferredNameDataDefinition(), formatter);
+
+    pdd.setParameters(getParameters());
+
+    pdd.addRowFilter(
+        listOfPatientsWithHighViralLoadCohortQueries.getPatientsWithUnsuppressedVlResult(),
+        "startDate=${startDate},endDate=${endDate},location=${location}");
+    pdd.addColumn("id", new PersonIdDataDefinition(), "");
+    // 1- NID sheet 1 - Column A
+    pdd.addColumn(
+        "nid",
+        tptListOfPatientsEligibleDataSet.getNID(identifierType.getPatientIdentifierTypeId()),
+        "");
+
+    // 2 - Name - Sheet 1: Column B
+    pdd.addColumn("name", nameDef, "");
+
+    // 3 - Idade - Sheet 1: Column C
+    pdd.addColumn(
+        "age",
+        listOfPatientsArtCohortCohortQueries.getAge(),
+        "evaluationDate=${evaluationDate}",
```

Comment:
Use endDate instead of evaluationDate to calculate the age **FR8**
Make the same change on SemanaCorrenteDataset